src/rules/__tests__/no-untyped-mock-factory.test.ts

```diff
@@ -0,0 +1,137 @@
+import path from 'path';
+import { ESLintUtils } from '@typescript-eslint/utils';
+import dedent from 'dedent';
+import rule from '../no-untyped-mock-factory';
+
+function getFixturesRootDir(): string {
+  return path.join(__dirname, 'fixtures');
+}
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    sourceType: 'module',
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-untyped-mock-factory', rule, {
+  valid: [
```

Comment:
Could you add a test for `jest.mock<typeof import('foo')>('bar', ...)`, to showcase that we don't check the actual import path?